src/ch/imedias/rsccfx/view/RsccSupportPresenter.java

```diff
@@ -61,45 +58,54 @@ public void initSize(Scene scene) {
   }
 
   /**
-   * Validates the token and displays a symbolic image.
-   *
-   * @param token the token to be validated.
-   * @return path to the image to display.
+   * Sets the token validation image depending on the validity of the token.
    */
-  public Image validationImage(String token) {
-
-    if (validateToken(token)) {
-      view.connectBtn.setDisable(false);
-      return new Image(getClass().getClassLoader().getResource("emblem-default.png")
-          .toExternalForm());
-    }
-    view.connectBtn.setDisable(true);
-    return new Image(getClass().getClassLoader().getResource("dialog-error.png")
-        .toExternalForm());
+  public void setValidationImage(boolean isValid) {
+    String imageFileName = isValid ? VALID_IMAGE_FILENAME : INVALID_IMAGE_FILENAME;
+    view.isValidImg.setImage(
+        new Image(getClass().getClassLoader().getResource(imageFileName).toExternalForm())
+    );
   }
 
   /**
    * Updates the validation image after every key pressed.
    */
   private void attachEvents() {
-
-    view.tokenFld.setOnKeyReleased(event -> {
-      view.isValidImg.setImage(validationImage(view.tokenFld.getText()));
-    });
-
+    view.connectBtn.disableProperty().addListener(
+        (observable, oldValue, newValue) -> {
+          // value = whether the button is disabled or not
```

Comment:
To which value are you referring to? oldValue, newValue? I'd change it in the comment itself to make it clear